src/AutoRest.CSharp/Common/Output/Models/Shared/Parameter.cs

```diff
@@ -20,6 +20,7 @@ internal record Parameter(string Name, FormattableString? Description, CSharpTyp
     {
         public CSharpAttribute[] Attributes { get; init; } = Array.Empty<CSharpAttribute>();
         public bool IsOptionalInSignature => DefaultValue != null;
+        public bool IsEndpoint => Type.Equals(typeof(Uri)) && Name == KnownParameters.Endpoint.Name;
```

Comment:
this does not look like correct - we have a `IsEndpoint` flag in `InputParameter`, and I think if we have this, this should have the same value as that flag.
https://github.com/Azure/autorest.csharp/blob/2755f061b8f45ce0017a3b61c949eba06d1ff613/src/AutoRest.CSharp/Common/Input/InputTypes/InputParameter.cs#L24

If there is inconsistency when it comes to the swagger inputs, we might need to refine the converter